### PR TITLE
add validations and model specs

### PR DESCRIPTION
**Pull request by fickryiman**
- Add the following validations: 
     - For the User model: 
     - Name must not be blank. 
     - PostsCounter must be an integer greater than or equal to zero. 
  - For the Post model: 
     - The title must not be blank. 
     - The title must not exceed 250 characters. 
     - CommentsCounter must be an integer greater than or equal to zero. 
     - LikesCounter must be an integer greater than or equal to zero.
  - Add unit specs for all of the models' methods and validations.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -3,6 +3,10 @@ class Post < ApplicationRecord
   has_many :comments, foreign_key: :post_id
   has_many :likes, foreign_key: :post_id
 
+  validates :title, presence: true, length: { maximum: 250 }
+  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
   def recent_comments
     comments.order(created_at: :desc).limit(5)
   end
```
Comment (elisha2kyakpo1) [inline app/models/post.rb:14]: - [x] Kindly test this function and the one that __updates the posts counter__ as it is a requirement to test all the model methods.

File: app/models/user.rb [hunk 2]
```diff
@@ -3,6 +3,9 @@ class User < ApplicationRecord
   has_many :likes, foreign_key: :author_id
   has_many :comments, foreign_key: :author_id
 
+  validates :name, presence: true
+  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
   def recent_posts
     posts.order(created_at: :desc).limit(3)
   end
```
Comment (elisha2kyakpo1) [inline app/models/user.rb:11]: - [x] Kindly test this function to see if it can return the 3 most recent posts.

File: spec/like_spec.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+RSpec.describe Like, type: :model do
+  it 'is not valid without an author_id' do
+    like = Like.new(post_id: 1)
+    expect(like).to_not be_valid
+  end
+
+  it 'is not valid without a post_id' do
+    like = Like.new(author_id: 1)
+    expect(like).to_not be_valid
+  end
```
Comment (elisha2kyakpo1) [inline spec/like_spec.rb:12]: - [x] The function that updates the likes counter is not covered in the tests, kindly test it.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @fickryiman, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file👏
👏 Model validations are set up nicely.

## Required Changes ♻️
_Check the comments under the review._

- [x] The live link in the demo section is not correct, kindly provide a correct link or remove the whole section.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/like_spec.rb (hunk 3 above)
Comment (prolajumokeoni) [reply spec/like_spec.rb:12]: Hi @fickryiman, this test is not yet added 👍🏿

File: app/models/user.rb (hunk 2 above)
Comment (prolajumokeoni) [reply app/models/user.rb:11]: Hi @fickryiman, this test is not yet added 👍🏿

File: spec/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  it 'is not valid with a non-integer posts_counter' do
+    user = User.new(name: 'user-1', posts_counter: 2.5)
+    expect(user).to_not be_valid
+  end
+
+  it 'is not valid with a negative posts_counter' do
+    user = User.new(name: 'user-1', posts_counter: -1)
+    expect(user).to_not be_valid
+  end
+end
```
Comment (ghost) [inline spec/user_spec.rb]: - [x] As requested earlier, Kindly test the `recent_posts` method in your user model

File: spec/post_spec.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    post = Post.new(text: 'post text', author:, comments_counter: 0, likes_counter: 0)
+    expect(post).to_not be_valid
+  end
+
+  it 'is not valid when a title exceed 250 characters' do
+    author = User.create(name: 'author-1')
+    title = 'a' * 251
+    post = Post.new(title:, text: 'post text', author:, comments_counter: 0, likes_counter: 0)
+    expect(post).to_not be_valid
+  end
```
Comment (ghost) [inline spec/post_spec.rb:21]: - [x] Kindly also test these methods, `recent_comments`  and `update_posts_counter` in your post model to ensure your app behaves as expected

**Review by ghost — CHANGES_REQUESTED** [review]
Hi @fickryiman , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights

- [x] Model validations are setup properly


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by aamirkhan2478 — APPROVED** [review]
Hi @fickryiman, 

GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ You used Gitflow correctly :+1: 
✅ No linter error :+1: 
✅ Good commit messages :+1: 
✅ Good descriptive PR :+1: 
✅ Well done :clap: you did all changes given by the previous reviewers :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**